controller/cmd/service-mirror/cluster_watcher.go

```diff
@@ -26,14 +27,37 @@ type (
 	// it can be requeued up to N times, to ensure that the failure is not due to some temporary network
 	// problems or general glitch in the Matrix.
 	RemoteClusterServiceWatcher struct {
-		clusterName     string
-		clusterDomain   string
-		remoteAPIClient *k8s.API
-		localAPIClient  *k8s.API
-		stopper         chan struct{}
-		log             *logging.Entry
-		eventsQueue     workqueue.RateLimitingInterface
-		requeueLimit    int
+		clusterName        string
+		clusterDomain      string
+		remoteAPIClient    *k8s.API
+		localAPIClient     *k8s.API
+		stopper            chan struct{}
+		log                *logging.Entry
+		eventsQueue        workqueue.RateLimitingInterface
+		requeueLimit       int
+		probePort          int32
+		probePath          string
+		probePeriodSeconds int32
+		enqueueProbeEvent  func(event interface{})
```

Comment:
It might be more clear and concrete to hold a `*ProbeManager` instead of a totally generic `func(event interface{})`.  I don't think the generality really gets us anything useful since we only ever enqueue to a ProbeManager.

Reply:
I wanted to keep it like that for test purposes. Namely., i want to exercise a bunch of events coming from the informer and expect that a bunch of other events will be send to the ProbeManager. And in tests, I think it might be cleaner to just provide a function that buffers them rather than mocking the ProbeManager (for some definition of Mocking that Go adopts... )

Reply:
What do you think about creating a `ProbeEventSink` interface which defines just a `send` method?  This is basically exactly what you already have, but using an interface feels a bit more like idiomatic go than passing around an anonymous function.  TIOLI.